File: frontend/src/types/coupon.ts
export interface CouponListItemDTO {
  couponId: number
  serialNumber?: string
  eventId?: number | null
  couponCategoryId?: number | null
  eventEligibilityId?: number | null
  couponName: string
  couponDescription?: string
  couponCategory?: string | null
  discountAmount?: number | null
  minimumSpend?: number | null
  status: '已領取' | '未領取'
  redeemableTimes?: number | null
  quantity?: number | null
  couponImageBase64?: string
}

export interface MemberCouponListDTO {
  memberCouponId: number
  memberId: number
  couponId: number
  status: '未使用' | '已使用'
  acquisitionTime?: string
  usageTime?: string
  serialNumber?: string
  couponName: string
  couponDescription?: string
  couponCategory?: string | null
  discountAmount?: number | null
  minimumSpend?: number | null
  couponImageBase64?: string
}

export interface PageResp<T> {
  content: T[]
  totalElements: number
  number: number // backend page index (1-based in our API)
  size: number
}

File: backend/src/main/java/com/flux/movieproject/model/entity/event/Coupon.java
package com.flux.movieproject.model.entity.event;

import com.fasterxml.jackson.annotation.JsonIdentityReference;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.flux.movieproject.enums.DiscountType;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Getter
@Setter
@NoArgsConstructor
@Table
public class Coupon {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer couponId;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "event_id", nullable = false)
    @JsonIdentityReference(alwaysAsId = true)
    @JsonIgnoreProperties({"hibernateLazyInitializer","handler"})
    private Event event;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "event_eligibility_id")
    private EventEligibility eventEligibility;

    private String serialNumber;
    private String couponName;
    private String couponDescription;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "coupon_category_id")
    private CouponCategory couponCategory;
    
    @Enumerated(EnumType.STRING) 
	private DiscountType discountType;

    private Integer discountAmount;
    private Integer minimumSpend;

    @Lob
    private byte[] couponImage;

    private String status;
    private Integer redeemableTimes;
    private Integer quantity;
}
